da/celestia/celestia.go

```diff
@@ -111,60 +109,92 @@ func (c *DataAvailabilityLayerClient) SubmitBlock(ctx context.Context, block *ty
 }
 
 // CheckBlockAvailability queries DA layer to check data availability of block at given height.
-func (c *DataAvailabilityLayerClient) CheckBlockAvailability(
-	ctx context.Context, dataLayerHeight uint64,
-) da.ResultCheckBlock {
-	shares, err := c.client.NamespacedShares(ctx, c.namespaceID, dataLayerHeight)
-	code := dataRequestErrorToStatus(err)
-	if code != da.StatusSuccess {
+func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
+	header, err := c.rpc.Header.GetByHeight(ctx, dataLayerHeight)
+	if err != nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+	if header.DAH == nil {
+		return da.ResultCheckBlock{
+			BaseResult: da.BaseResult{
+				Code:     da.StatusSuccess,
+				DAHeight: dataLayerHeight,
+			},
+			DataAvailable: false,
+		}
+	}
+	err = c.rpc.Share.SharesAvailable(ctx, header.DAH)
+	if err != nil {
 		return da.ResultCheckBlock{
 			BaseResult: da.BaseResult{
 				Code:    da.StatusError,
 				Message: err.Error(),
 			},
 		}
 	}
-
 	return da.ResultCheckBlock{
 		BaseResult: da.BaseResult{
 			Code:     da.StatusSuccess,
 			DAHeight: dataLayerHeight,
 		},
-		DataAvailable: len(shares) > 0,
+		DataAvailable: true,
```

Comment:
This should be `DataAvailable: err.Is(share.ErrNotAvailable)`

Reply:
`err.Is` won't work unless the error object is same. celestia-node `ErrNotAvailable` is not same as celestia-openrpc `ErrNotAvailable`. you may have to compare the Error() strings.

Reply:
Would be ideal if celestia reuses rollkit type.

Reply:
Checking error string, closed linked PR.